go/k8s/api/v1/user_types.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// UserSpec defines the desired state of User
+type UserSpec struct {
+	ID   int64  `json:"id,omitempty"`
+	Mode string `json:"mode,omitempty"`
```

Comment:
Is there a way to validate that "mode" and "state" have the allowed values? Would that be in the CRD? Where is the CRD, by the way?

Reply:
Please check the PR in sysadmin repo.  Validator for the specs can be added later.